pkg/executor/join/hash_join_v2.go

```diff
@@ -1047,6 +1047,27 @@ func (e *HashJoinV2Exec) reset() {
 	}
 }
 
+func (e *HashJoinV2Exec) collectSpillStats() {
+	if e.stats == nil || !e.spillHelper.isSpillTriggered() {
+		return
+	}
+
+	round := e.spillHelper.round
+	if len(e.stats.spill.totalSpillBytesPerRound) < round+1 {
```

Comment:
Will it be possible that length + N < round here? If so, for loop makes sense

Reply:
> Will it be possible that length + N < round here? If so, for loop makes sense

It's impossible.